### PR TITLE
[SPARK-25167][SPARKR][TEST][MINOR] Minor fixes for R sql tests (timestamp comparison)

### PR DESCRIPTION
**Pull request by dilipbiswal**
## What changes were proposed in this pull request?
The "date function on DataFrame" test fails consistently on my laptop. In this PR
i am fixing it by changing the way we compare the two timestamp values. With this change i am able to run the tests clean.

## How was this patch tested?
Fixed the failing test.

### CODE REVIEW

File: R/pkg/tests/fulltests/test_sparkSQL.R [hunk 1]
```diff
@@ -3633,7 +3633,8 @@ test_that("catalog APIs, currentDatabase, setCurrentDatabase, listDatabases", {
   expect_equal(currentDatabase(), "default")
   expect_error(setCurrentDatabase("default"), NA)
   expect_error(setCurrentDatabase("zxwtyswklpf"),
-        "Error in setCurrentDatabase : analysis error - Database 'zxwtyswklpf' does not exist")
+               paste("Error in setCurrentDatabase : analysis error - Database",
```
Comment (dilipbiswal) [inline R/pkg/tests/fulltests/test_sparkSQL.R]: @HyukjinKwon here i am fixing the last un-addressed [comment](https://github.com/apache/spark/pull/22161#discussion_r212168906)

**Comment by HyukjinKwon** [comment]
> The "date function on DataFrame" test fails consistently on my laptop.

@dilipbiswal, would you mind if I ask to list up R version (and additional env information if possible)?

**Review by HyukjinKwon — APPROVED** [review]
Looks right. cc @felixcheung as well

**Comment by dilipbiswal** [comment]
@HyukjinKwon Sure.. here is the info. Please let me know if you need anything else.

```
> sessionInfo()
R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.3

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] testthat_1.0.2

loaded via a namespace (and not attached):
[1] compiler_3.5.1 magrittr_1.5   R6_2.2.2       crayon_1.3.4 
```

File: R/pkg/tests/fulltests/test_sparkSQL.R (hunk 1 above)
Comment (felixcheung) [reply R/pkg/tests/fulltests/test_sparkSQL.R]: I'd use paste0 instead to make clear about the implicit space that should be after `Database`

ie. `paste0("Error in setCurrentDatabase : analysis error - Database ",  "'zxwtyswklpf' does not exist"))

Comment (dilipbiswal) [reply R/pkg/tests/fulltests/test_sparkSQL.R]: @felixcheung Sure.

Comment (dilipbiswal) [reply R/pkg/tests/fulltests/test_sparkSQL.R]: @felixcheung Made the change to use paste0. Please take a look when you get a chance. Thanks !!

**Comment by felixcheung** [comment]
maybe also your laptop's system time zone? could you also check that?

**Comment by felixcheung** [comment]
interesting. maybe something to do with newer R release - I scanned through the rel note though but didn't find what might be related.

**Comment by felixcheung** [comment]
possible - but since this passes for you and in jenkins/appveyor you change seem to work both ways, which is good enough for me

**Review by felixcheung — APPROVED** [review]

**Comment by felixcheung** [comment]
merged to master

**Comment by dilipbiswal** [comment]
@felixcheung Thank you very much !!

**Pull request closed without merging**